vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -164,6 +174,34 @@ void AdsImpl::InitializeStep4(
   ads_serve_->DownloadCatalog();
 }
 
+#if defined(OS_ANDROID)
+void AdsImpl::RemoveAllNotificationsAfterReboot() {
+  //ads notifications don't sustain reboot, so remove all
+  auto ads_shown_history = client_->GetAdsShownHistory();
+  if (!ads_shown_history.empty()) {
```

Comment:
please fail early so if empty return as then we keep away from the pyramid of doom

Reply:
There is no pyramid or nesting. The code is clear and explicit.

Reply:
NONCRITICAL
Covered by: brave/brave-browser#6235 (1)